src/objects/objects.cc

```diff
@@ -5607,6 +5602,10 @@ MaybeHandle<Object> JSPromise::Resolve(Handle<JSPromise> promise,
     // is intact, as that guards the lookup path for the "then" property
     // on JSPromise instances which have the (initial) %PromisePrototype%.
     then = isolate->promise_then();
+
+    if (RecordReplayShouldCallOnPromiseHook() && resolution->IsJSReceiver()) {
+      AddPromiseDependencyGraphAdoption(isolate, promise, Handle<Object>::cast(resolution));
+    }
```

Comment:
2 notes here:
- this is currently only called when adopting native promises. I have no idea how thenables work today but we should, ideally, support them too. It might require an extra investigation into them to see how well we can support them. Are they ever converted to promises internally? If yes then that would be great but that also might mean this should get instrumented elsewhere
- even before this proposed change the new information was added only this function (`JSPromise::Resolve`), what about `JSPromise::Reject` though? It feels the instrumentation is missing there but I can't really be sure. If that's the case then perhaps we should look into instrumenting `TriggerPromiseReactions`?

Reply:
but tbf I'm also wondering if this instrumentation site isn't "too late". It feels like this would never create the `adoptedPromise` edge in a case like this:
```ts
const neverResolves = () => new Promise(() => {})

async function test() {
  await neverResolves() 
}
```

And that it would be better to have that edge.

Reply:
Although, I guess the promise adoption actually only happens for successful resolutions. So `JSPromise::Reject` and stuff is irrelevant here. And this might be the proper place to instrument this. Once this whole thing lands in one shape or another we should add a test case like:
```ts
function timeout(ms) {
  return new Promise((_, reject) => setTimeout(() => reject(new Error('Timeout.'), ms))
}
function test() {
  try {
    await timeout(10)
  } catch (err) {
    // point
  }
}
```

Reply:
* I changed it to be called on all adopted values without checking for native promises.
* Instead, I checked for whether the value did have a `newPromise` event in `AddPromiseDependencyGraphAdoption`, and simply omit it if not.
* `TriggerPromiseReactions` does not have access to the `promise` itself, but we can move it right next to its two call sites.